src/bigquery.ts

```diff
@@ -245,9 +245,12 @@ export class BigQuery extends common.Service {
 
   constructor(options: BigQueryOptions = {}) {
     options.apiEndpoint = options.apiEndpoint || 'bigquery.googleapis.com';
+    const url =
```

Comment:
@tswast @shollyman  this feels like something we should do consistently across clients, if we want to do it.  What do we do in other languages?

Reply:
This was already completed in Python via the "client options" property. go/extensible-options-in-client

Reply:
Hmm... Not seeing a design for Node.js in that internal doc, though. Just Go & Python, but it definitely needs to be done across all languages.

Reply:
@JustinBeckwith would this require a design change for the library. We are about to migrate our CI pipeline to cloud build and were trying to avoid the route of using the SAN certs for our integration tests.

Reply:
This does come up occasionally for all of our emulators. I wonder if we could introduce into the auth library, `EMULATOR_HOST`.

Reply:
This does come up occasionally for all of our emulators. I wonder if we could introduce into the auth library, `EMULATOR_HOST`.

Reply:
This does come up occasionally for all of our emulators. I wonder if we could introduce into the auth library, `EMULATOR_HOST`.